File: typescript/difference-of-squares/difference-of-squares.ts
class Squares {
  constructor(private _n: number) {}

  public get squareOfSum(): number {
    let sum: number = 0;

    for (let i = 1; i <= this._n; i++) {
      sum += i;
    }

    return sum ** 2;
  }

  public get sumOfSquares(): number {
    let sum: number = 0;

    for (let i = 1; i <= this._n; i++) {
      sum += i ** 2;
    }

    return sum;
  }

  public get difference(): number {
    return this.squareOfSum - this.sumOfSquares;
  }
}

export default Squares;

File: kotlin/difference-of-squares/src/main/kotlin/DifferenceOfSquares.kt
import kotlin.math.pow

class Squares(val n: Int) {

    fun sumOfSquares(): Int {
        return (1..n).sumOf { it * it }
    }

    fun squareOfSum(): Int {
        return (1..n).sum().toDouble().pow(2).toInt()
    }

    fun difference(): Int {
        return squareOfSum() - sumOfSquares()
    }
}
